cmd/apps/traefik2_app.go

```diff
@@ -0,0 +1,172 @@
+package apps
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/helm"
+	execute "github.com/alexellis/go-execute/pkg/v1"
+	"github.com/morikuni/aec"
+	"github.com/spf13/cobra"
+)
+
+func installTraefik2(parts ...string) (execute.ExecResult, error) {
+
+	task := execute.ExecTask{
+		Command:     "helm",
+		Args:        parts,
+		StreamStdio: true,
+	}
+	res, err := task.Execute()
+	if err != nil {
+		return res, err
+	}
+	if res.ExitCode != 0 {
+		return res, fmt.Errorf("exit code %d, stderr: %s", res.ExitCode, res.Stderr)
+	}
+	return res, nil
+}
+
+func MakeInstallTraefik2() *cobra.Command {
+	var traefik2 = &cobra.Command{
+		Use:          "traefik2",
+		Short:        "Install traefik2",
+		Long:         "Install traefik2",
+		Example:      `  arkade app install traefik2`,
+		SilenceUsage: true,
+	}
+
+	var dashboard bool
+	traefik2.Flags().StringP("namespace", "n", "kube-system", "The namespace used for installation")
+	traefik2.Flags().Bool("update-repo", true, "Update the helm repo")
+	traefik2.Flags().BoolVar(&dashboard, "dashboard", false, "Expose dashboard if you want access to dashboard from the browser")
+	traefik2.Flags().StringArray("set", []string{},
+		"Use custom flags or override existing flags \n(example --set key=value)")
+
+	traefik2.RunE = func(command *cobra.Command, args []string) error {
+		PrintTraefikASCIIArt()
+		kubeConfigPath := getDefaultKubeconfig()
+		if command.Flags().Changed("kubeconfig") {
+			kubeConfigPath, _ = command.Flags().GetString("kubeconfig")
+		}
+		fmt.Printf("Using kubeconfig: %s\n", kubeConfigPath)
+
+		updateRepo, _ := traefik2.Flags().GetBool("update-repo")
+		namespace, _ := traefik2.Flags().GetString("namespace")
+		userPath, err := config.InitUserDir()
+		if err != nil {
+			return err
+		}
+
+		_, clientOS := env.GetClientArch()
+		clientArch, clientOS := env.GetClientArch()
+		fmt.Printf("Client: %q\n", clientOS)
+
+		_, err = helm.TryDownloadHelm(userPath, clientArch, clientOS, false)
+		if err != nil {
+			return err
+		}
+
+		err = addHelmRepo("traefik", "https://containous.github.io/traefik-helm-chart", false)
+		if err != nil {
+			return fmt.Errorf("Unable to add repo %s", err)
+		}
+
+		if updateRepo {
+			err = updateHelmRepos(false)
+			if err != nil {
+				return err
+			}
+		}
+
+		chartPath := path.Join(os.TempDir(), "charts")
+		err = fetchChart(chartPath, "traefik/traefik", "", false)
+		if err != nil {
+			return fmt.Errorf("Unable fetch char %s", err)
+		}
+
+		overrides := map[string]string{}
+		overrides["service.type"] = "NodePort"
```

Comment:
Maybe this should be an an Option and defaults to NodePort or point out in the message how to change it.

Reply:
It should be a LoadBalancer normally, that's what k3s does for Traefik 1.x